### PR TITLE
api: add LoginTest to avoid regression of #2549

### LINKED ISSUE #2556: Write an API test for loading a user that does not exist
Follow-up from #2549, we should write a test for this bug, so that no regression can happen.

### PR DESCRIPTION
**Pull request by BacLuc**
And also add not found tests to ReadUserTest.
closes #2556

### CODE REVIEW

File: api/tests/Api/Users/ReadUserTest.php [hunk 1]
```diff
@@ -19,6 +19,18 @@ public function testGetSingleUserIsDeniedForAnonymousUser() {
         ]);
     }
 
+    public function testGetUnknownUserReturns404ForAnonymousUser() {
+        static::createClientWithCredentials()->request('GET', '/users/1');
```
Comment (carlobeltrame) [inline api/tests/Api/Users/ReadUserTest.php]: "With credentials" means this API call is not anonymous.

**Review by carlobeltrame — CHANGES_REQUESTED** [review]
Oops, I should have selected "request changes"

File: api/tests/Api/Users/ReadUserTest.php (hunk 1 above)
Comment (BacLuc) [reply api/tests/Api/Users/ReadUserTest.php]: Sorry, typical case of "I will get this done quickly before my next appointment." :cry:

Comment (carlobeltrame) [reply api/tests/Api/Users/ReadUserTest.php]: That's why we do code reviews :)

**Review by carlobeltrame — APPROVED** [review]
Thanks!

**Review by usu — APPROVED** [review]

**Pull request merged**